source/shared/content/_archive.slim

```diff
@@ -1,6 +1,14 @@
 h2 Archive
-ul
-  -past_events.each_with_index do |past_event|
-    li
-      a href="#"
-        ="#{past_event.date}"
+
+- #event.date[0,4] is event.date.year
+-grouped_events = past_events.group_by{ |event| event.date[0,4] }.values
+-grouped_events.each do |a_year_events|
```

Comment:
important:  this is *only a suggestion*.  ultimately you decide what makes most sense to you.